udf_js/json_extract_missing_cols.sql

```diff
@@ -0,0 +1,90 @@
+/*
+
+Extract missing columns from additional properties.
+
+More generally, get a list of nodes from a JSON blob.
+
+param input: The JSON blob to explode
+param indicates_node: An array of strings. If a key's value is an object, and contains one of these values,
+                      that key is returned as a node.
+param known_nodes: An array of strings. If a key is in this array, it is returned as a node.
+
+Notes:
+- Use indicates_node for things like histograms. For example ['histogram_type'] will ensure that each
+  histogram will be returned as a missing node, rather than the subvalues within the histogram
+  (e.g. values, sum, etc.)
+- Use known_nodes if you're aware of a missing section, like ['simpleMeasurements']
+
+See here for an example usage
+https://sql.telemetry.mozilla.org/queries/64460/source
+ 
+*/
+
+CREATE TEMP FUNCTION udf_js_json_extract_missing_cols (input STRING, indicates_node ARRAY<STRING>, known_nodes ARRAY<STRING>)
```

Comment:
This function does not handle arrays of structs well, but that's not a common use case.